packages/superset-ui-core/src/query/processFilters.ts

```diff
@@ -5,53 +5,46 @@ import { isSimpleAdhocFilter } from './types/Filter';
 import convertFilter from './convertFilter';
 
 /** Logic formerly in viz.py's process_query_filters */
-export default function processFilters(formData: QueryFormData) {
+export default function processFilters(formData: QueryFormData): Partial<QueryFormData> {
   // Split adhoc_filters into four fields according to
   // (1) clause (WHERE or HAVING)
   // (2) expressionType
   //     2.1 SIMPLE (subject + operator + comparator)
   //     2.2 SQL (freeform SQL expression))
-  const { adhoc_filters } = formData;
-  if (Array.isArray(adhoc_filters)) {
-    const simpleWhere: QueryObjectFilterClause[] = formData.filters || [];
-    const simpleHaving: QueryObjectFilterClause[] = [];
-    const freeformWhere: string[] = [];
-    if (formData.where) freeformWhere.push(formData.where);
-    const freeformHaving: string[] = [];
+  const { adhoc_filters = [], extras = {}, filters = [], where } = formData;
+  const simpleWhere: QueryObjectFilterClause[] = filters;
```

Comment:
Changes to this file are mostly unrelated, but the cleanup was done when native filtering logic was thought to be added to `processFilters.ts`. This logic was however moved to a utility of it's own, but the refactoring was left as-is.